service/controller/clusterapi/v16/resources/awsclusterconfig/create.go

```diff
@@ -20,7 +24,82 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return nil
 	}
 
-	// TODO: Map Cluster -> AWSClusterConfig.
+	// Get existing AWSClusterConfig or create new one.
+	awsClusterConfig, err := r.getAWSClusterConfig(ctx, cluster)
```

Comment:
Wouldn't it be simpler to query for the object in here and have some logic based on the error? I think we did it like this in other resources. The check below about the object "generation" is a bit magical.

Reply:
I wrote it like that initially, but it made this `EnsureCreated()` function longer and more difficult to follow and that's why I changed it like this. This way there doesn't need to be so much repetition. I don't know if generation is really magical. I can add comment for it if it helps?

Reply:
That generation check is super magical because I do not get it. The intention is to rely on the version to be 0 when the resource is freshly created or when there is the zero value returned which then implies no existence? I think that is super confusing.